pkg/messaging/v1/invoke_method_request.go

```diff
@@ -171,6 +178,15 @@ func (imr *InvokeMethodRequest) RawData() (string, []byte) {
 	contentType := m.GetContentType()
 	dataValue := m.GetData().GetValue()
 
+	// TODO: Remove once feature is finalized
+	if !imr.NoDefaultContentType {
+		dataTypeURL := m.GetData().GetTypeUrl()
+		// set content_type to application/json only if typeurl is unset and data is given
+		if contentType == "" && (dataTypeURL == "" && dataValue != nil) {
+			contentType = JSONContentType
```

Comment:
Can you add a test that covers this? Also, was this the original behavior?

Reply:
That was the original behavior, see: https://github.com/dapr/dapr/pull/4472/files#diff-5be38c56c52875424fffeea0582f2d3dc58c2e801f22e49f5a0f85d5c6600129L178-L182

I will add a test